lib/client.js

```diff
@@ -298,7 +297,9 @@ class Client extends EventEmitter {
       delete tarnPoolConfig.afterCreate;
     }
 
-    this.pool = new Pool(tarnPoolConfig);
+    const pool = new Pool(tarnPoolConfig);
+    tracker.addPool(pool);
```

Comment:
This essentially just binds some event handlers that add/remove entries when Tarn reports that connections are acquired or released. Tarn's docs specify that these event handlers are synchronous, and I verified experimentally that the event handlers run before e.g. `pool.acquire()` returns -- so we can be assured that the connection returned from Tarn has an id assigned to it by the time we first have a reference to it